composer/helpers/v1/src/UpdateChecker.php

```diff
@@ -82,7 +82,16 @@ public static function getLatestResolvableVersion(array $args): ?string
 
         // We found the package in the list of updated packages. Return its version.
         if ($updatedPackage instanceof PackageInterface) {
-            return preg_replace('/^([v])/', '', $updatedPackage->getPrettyVersion());
+            // TODO surprisingly the returned result of getPrettyVersion depends on the PHP version:
+            // - PHP 7 returns: "2.4.1"
+            // - PHP 8 returns: "2.4.1@stable"
+            // file_put_contents('php://stdout', $updatedPackage->getPrettyVersion());
+            //
+            // return preg_replace('/^([v])/', '', $updatedPackage->getPrettyVersion());
+
+            $pretty = $updatedPackage->getPrettyVersion();
+
+            return rtrim(ltrim($pretty, 'v'), '@stable');
```

Comment:
@Seldaek any idea why the returned value of `getPrettyVersion()` version depends on the underlying PHP version? I found this very surprising given that the composer code didn't change, only the php version.

This is a unit test running under `composer` `1`, accessing a PEAR registry.

I do realize that both PEAR and composer v1 are deprecated, but we're not _quite_ ready to drop support for them, so I'd like to fix our code to buy us compatibility for a little longer.

And really what I care about is:
Is this `@stable` suffix always going to be the same returned value such that it's safe for me to simply strip it off? Or could it sometimes return `@beta`, `-dev`, etc? 

IE, is `@stable` something PHP itself is adding in some manner in which case I'd always expect it to be there and I can safely strip it (I don't know if PHP has a concept of stable/unstable objects aka immutable?), or is it something `composer` is adding, so it may vary depending on the package/registry?


Note to myself - encountered while running:
```shell
[dependabot-core-dev] ~/dependabot-core/composer $ DEBUG_HELPERS=true rspec ./spec/dependabot/composer/update_checker_spec.rb:451
```

Reply:
nudge @Seldaek, any chance of quick input on ☝️ ?

To be clear, not asking for you do to additional research, just if you know off top of your head.

This is the only remaining unresolved question for this PR to get Dependabot to support PHP 8.